app/models/grant.rb

```diff
@@ -60,11 +66,46 @@ class Grant < ApplicationRecord
 
   validate :valid_default_set, on: :create, unless: -> { duplicate.present? }
 
+  before_destroy :deletable?
+
   scope :by_publish_date,    -> { order(publish_date: :asc) }
   scope :with_organization,  -> { joins(:organization) }
   scope :with_questions,     -> { includes :questions }
 
+  def is_soft_deletable?
+    SOFT_DELETABLE_STATES.include?(state) ? true : send("#{state}_soft_deletable?")
+  end
+
+  private
+
+  def deletable?
+    # TODO: Review destroy / soft delete logic as other models are added
+    raise SoftDeleteException.new('Grants must be soft deleted.')
+  end
+
   def valid_default_set
     errors.add(:base, 'Please choose a default question set') unless DefaultSet.where(id: default_set).exists?
   end
+
+  def published_soft_deletable?
```

Comment:
Do we still these individual soft_deletable? method? I think general is_soft_deletable? should be good enough.

Reply:
There will be some logic on a published grant -- e.g. has there been a submission